frontend/services/opengl_glfw/OpenGL_GLFW_Service.hpp

```diff
@@ -52,32 +52,39 @@ class OpenGL_GLFW_Service final : public AbstractFrontendService {
     std::string serviceName() const override { return "OpenGL_GLFW_Service"; }
 
     OpenGL_GLFW_Service() = default;
-    ~OpenGL_GLFW_Service() override;
+    ~OpenGL_GLFW_Service() override {}
     // TODO: delete copy/move/assign?
 
+    void resetProvidedResources() override {
+        m_keyboardEvents.clear();
+        m_mouseEvents.clear();
+        m_windowEvents.clear();
+        m_framebufferEvents.clear();
+    }
+
+    void preGraphRender() override {}  // prepare rendering with API, e.g. set OpenGL context, frame-timers, etc
+    void postGraphRender() override {
+        this->do_every_second();
+    } // clean up after rendering, e.g. stop and show frame-timers in GLFW window
```

Comment:
I would like to have the non-gl function implementations in a cpp too. With implementations the header seems too crowded.